src/hotspot/share/compiler/compileBroker.cpp

```diff
@@ -2374,7 +2374,7 @@ void CompileBroker::invoke_compiler_on_method(CompileTask* task) {
     if (CompilationLog::log() != nullptr) {
       CompilationLog::log()->log_failure(thread, task, failure_reason, retry_message);
     }
-    if (PrintCompilation) {
+    if (PrintCompilation || task->directive()->PrintCompilationOption) {
```

Comment:
Sounds like a good idea, but is this the only place where we want to do something when either `-XX:+PrintCompilation` or `-XX:CompileCommand=printcompilation,*::*` is set? IOW, shouldn't other checks for `PrintCompilation` also take `-XX:CompileCommand=printcompilation,*::*` into account?

E.g. line 2182 above does:

```
  if (directive->PrintCompilationOption) {
    ResourceMark rm;
    task->print_tty();
  }
```

Shouldn't that be:

```
if (PrintCompilation || directive->PrintCompilationOption) {
   ...
}
```

Reply:
I guess there are also flags where we have this mismatch between global flag and compile command local flag. Maybe we should do a general pass over all the compile commands and check how we can align them better.

Reply:
Yes sounds good

Reply:
I filed [JDK-8352047](https://bugs.openjdk.org/browse/JDK-8352047).